### PR TITLE
Home work automation #10

### PR DESCRIPTION
**Pull request by TanyaB777**

### CODE REVIEW

**Comment by AntBelyi** [comment]
Please resolve conflicts

**Comment by AntBelyi** [comment]
Why there're only 3 tests? As far as I see you have more tests in PageObject hw, that should be changed to BDD approach

File: src/test/resources/features/HomePage.feature [hunk 1]
```diff
@@ -0,0 +1,20 @@
+Feature: Home Page Verification
+
+  Scenario: Verify Help Center page title
+    Given User is on the Home Page
+    When User clicks on the Help Center link
+    Then User is navigated to the Help Center page
```
Comment (AntBelyi) [inline src/test/resources/features/HomePage.feature]: Lets changes this step to more clear. For example: "Verify user is navigated to the Help Center page"

File: src/test/java/aqa_hw_10/step_definitions/HomePageStepDefinition.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+    @When("User clicks on the Help Center link")
+    public void clickHelpCenterLink() {
+        homePage.clickHelpCenterLink();
+    }
+
+    @Then("User is navigated to the Help Center page")
+    public void verifyHelpCenterPage() {
+        Assert.assertTrue(helpCenterPage.isOnHelpCenterPage(), "Not on Help Center page");
+    }
```
Comment (AntBelyi) [inline src/test/java/aqa_hw_10/step_definitions/HomePageStepDefinition.java]: As far as I see this verification not for Home Page. So, probably we need to create a separate step definitions class and move this method there?

File: src/test/resources/features/HomePage.feature [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    And the page title should be 'Найактуальніше'
```
Comment (AntBelyi) [inline src/test/resources/features/HomePage.feature]: Let's change this step "Verify page title is 'Найактуальніше' on the Help Center page"

File: src/test/java/aqa_hw_10/step_definitions/HomePageStepDefinition.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+    @And("the page title should be 'Найактуальніше'")
```
Comment (AntBelyi) [inline src/test/java/aqa_hw_10/step_definitions/HomePageStepDefinition.java]: This step definition is hardcoded only for "Найактуальніше" title and won't be able to reused for another step, for example for verification another titles. Let's changes it to "verify page title is {string} on Help Center page"

File: src/test/java/aqa_hw_10/step_definitions/HomePageStepDefinition.java [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+    @Then("User should see a page with title containing 'Кошик порожній'")
+    public void shouldSeePageWithTitleContaining() {
+        Assert.assertTrue(cartPage.getPageTitle().contains(EMPTY_CART_TEXT));
+    }
+
+    @And("User should see an image on the cart page")
+    public void shouldSeeAnImageOnTheCartPage() {
+        Assert.assertTrue(cartPage.imageIsPresent());
+    }
```
Comment (AntBelyi) [inline src/test/java/aqa_hw_10/step_definitions/HomePageStepDefinition.java]: Should be created another step definitions class for this step?

File: src/test/java/aqa_hw_10/step_definitions/HomePageStepDefinition.java [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+    @When("User clicks on the cart icon")
+    public void clickOnTheCartIcon() {
+        homePage.clickCartIcon();
+    }
+
+    @Then("User should see a page with title containing 'Кошик порожній'")
+    public void shouldSeePageWithTitleContaining() {
+        Assert.assertTrue(cartPage.getPageTitle().contains(EMPTY_CART_TEXT));
+    }
```
Comment (AntBelyi) [inline src/test/java/aqa_hw_10/step_definitions/HomePageStepDefinition.java]: Should be created another step definitions class for this step?

File: src/test/java/aqa_hw_10/step_definitions/HomePageStepDefinition.java [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+    @And("User clicks the track button")
+    public void clickTheTrackButton() {
+        packageTrackPage.clickPackageTrackButton();
+    }
+
+    @Then("User should see an error message")
+    public void shouldSeeAnErrorMessage() {
+        String actualMessage = packageTrackPage.getValidationMessage();
+        Assert.assertEquals(String.format(SEARCH_ERROR_MESSAGE, packageNumber), actualMessage);
+    }
```
Comment (AntBelyi) [inline src/test/java/aqa_hw_10/step_definitions/HomePageStepDefinition.java]: Should be created another step definitions class for this step?

File: src/test/java/aqa_hw_10/step_definitions/HomePageStepDefinition.java [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+
+    @And("User enters an invalid package number")
+    public void enterAnInvalidPackageNumber() {
+        packageTrackPage.typePackageNumberInput(packageNumber);
+    }
+
+    @And("User clicks the track button")
+    public void clickTheTrackButton() {
+        packageTrackPage.clickPackageTrackButton();
+    }
```
Comment (AntBelyi) [inline src/test/java/aqa_hw_10/step_definitions/HomePageStepDefinition.java]: Should be created another step definitions class for this step?

File: src/test/java/aqa_hw_10/step_definitions/HomePageStepDefinition.java [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+
+    @When("User navigates to the package tracking page")
+    public void navigateToPackageTrackingPage() {
+        homePage.clickPackageTrackLink();
+    }
+
+    @And("User enters an invalid package number")
+    public void enterAnInvalidPackageNumber() {
+        packageTrackPage.typePackageNumberInput(packageNumber);
+    }
```
Comment (AntBelyi) [inline src/test/java/aqa_hw_10/step_definitions/HomePageStepDefinition.java]: Should be created another step definitions class for this step?

File: src/test/java/aqa_hw_10/step_definitions/HomePageStepDefinition.java [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+
+    @And("User clicks the track button")
```
Comment (AntBelyi) [inline src/test/java/aqa_hw_10/step_definitions/HomePageStepDefinition.java]: This step is action. So, should be not "And" but "When"

File: src/test/java/aqa_hw_10/step_definitions/HomePageStepDefinition.java [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+    public void shouldSeeAnImageOnTheCartPage() {
+        Assert.assertTrue(cartPage.imageIsPresent());
+    }
+
+    @When("User navigates to the package tracking page")
+    public void navigateToPackageTrackingPage() {
+        homePage.clickPackageTrackLink();
+    }
+
+    @And("User enters an invalid package number")
```
Comment (AntBelyi) [inline src/test/java/aqa_hw_10/step_definitions/HomePageStepDefinition.java]: This step is action. So, should be not "And" but "When"

File: pom.xml [hunk 12]
```diff
@@ -43,5 +43,28 @@
...
+      <artifactId>junit</artifactId>
+      <version>4.13.2</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
+      <version>4.13.2</version>
+      <scope>test</scope>
+    </dependency>
```
Comment (AntBelyi) [inline pom.xml]: junit dependency is duplicated. Please remove the redundant dependencies

File: src/main/java/aqa_hw_10/CartPage.java [hunk 13]
```diff
@@ -0,0 +1,32 @@
+package aqa_hw_10;
+
+import aqa_hw_18.BasePage;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+
+import static com.codeborne.selenide.Condition.visible;
+import static com.codeborne.selenide.Selenide.$;
```
Comment (AntBelyi) [inline src/main/java/aqa_hw_10/CartPage.java:8]: Please remove unused imports

File: src/main/java/aqa_hw_10/CartPage.java [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+        return $(CART_IMAGE).shouldBe(visible).isDisplayed();
+    }
+
+    public boolean cartIsOpened() {
+        return $(CART_LIST).shouldBe(visible).isDisplayed();
+    }
+
+    public String getProductTitle() {
+        return $(CART_PRODUCT_TITLE).getText();
+    }
```
Comment (AntBelyi) [inline src/main/java/aqa_hw_10/CartPage.java]: Please remove unused methods

File: src/test/resources/features/HomePage.feature (hunk 3 above)
Comment (TanyaB777) [reply src/test/resources/features/HomePage.feature]: fixed

File: src/test/java/aqa_hw_10/step_definitions/HomePageStepDefinition.java (hunk 2 above)
Comment (TanyaB777) [reply src/test/java/aqa_hw_10/step_definitions/HomePageStepDefinition.java]: fixed

File: src/test/java/aqa_hw_10/step_definitions/HomePageStepDefinition.java (hunk 4 above)
Comment (TanyaB777) [reply src/test/java/aqa_hw_10/step_definitions/HomePageStepDefinition.java]: fixed

File: src/test/java/aqa_hw_10/step_definitions/HomePageStepDefinition.java (hunk 6 above)
Comment (TanyaB777) [reply src/test/java/aqa_hw_10/step_definitions/HomePageStepDefinition.java]: fixed

File: src/test/java/aqa_hw_10/step_definitions/HomePageStepDefinition.java (hunk 5 above)
Comment (TanyaB777) [reply src/test/java/aqa_hw_10/step_definitions/HomePageStepDefinition.java]: fixed

Comment (TanyaB777) [reply src/test/java/aqa_hw_10/step_definitions/HomePageStepDefinition.java]: fixed

File: src/test/java/aqa_hw_10/step_definitions/HomePageStepDefinition.java (hunk 8 above)
Comment (TanyaB777) [reply src/test/java/aqa_hw_10/step_definitions/HomePageStepDefinition.java]: fixed

File: src/test/java/aqa_hw_10/step_definitions/HomePageStepDefinition.java (hunk 7 above)
Comment (TanyaB777) [reply src/test/java/aqa_hw_10/step_definitions/HomePageStepDefinition.java]: fixed

File: src/test/java/aqa_hw_10/step_definitions/HomePageStepDefinition.java (hunk 11 above)
Comment (TanyaB777) [reply src/test/java/aqa_hw_10/step_definitions/HomePageStepDefinition.java]: fixed

File: src/test/java/aqa_hw_10/step_definitions/HomePageStepDefinition.java (hunk 10 above)
Comment (TanyaB777) [reply src/test/java/aqa_hw_10/step_definitions/HomePageStepDefinition.java]: fixed

File: src/main/java/aqa_hw_10/CartPage.java (hunk 13 above)
Comment (TanyaB777) [reply src/main/java/aqa_hw_10/CartPage.java:8]: fixed

File: pom.xml (hunk 12 above)
Comment (TanyaB777) [reply pom.xml]: fixed

File: src/main/java/aqa_hw_10/CartPage.java (hunk 13 above)
Comment (TanyaB777) [reply src/main/java/aqa_hw_10/CartPage.java:8]: fixed

File: src/main/java/aqa_hw_10/CartPage.java (hunk 14 above)
Comment (TanyaB777) [reply src/main/java/aqa_hw_10/CartPage.java]: fixed

**Comment by TanyaB777** [comment]
> Please resolve conflicts

resolved

**Comment by TanyaB777** [comment]
> Why there're only 3 tests? As far as I see you have more tests in PageObject hw, that should be changed to BDD approach

I did not have enough time

File: src/test/java/aqa_hw_10/step_definitions/HomePageStepDefinition.java (hunk 4 above)
Comment (AntBelyi) [reply src/test/java/aqa_hw_10/step_definitions/HomePageStepDefinition.java]: Not fixed!
There's still hardcoded value for "Найактуальніше" title. I've suggested you to change it to "verify page title is {string} on Help Center page"
Screenshot_4

File: pom.xml (hunk 12 above)
Comment (AntBelyi) [reply pom.xml]: not fully fixed

File: src/main/java/aqa_hw_10/CartPage.java (hunk 14 above)
Comment (AntBelyi) [reply src/main/java/aqa_hw_10/CartPage.java]: I've removed unused methods but haven't removed unused variables
Screenshot_5

File: src/test/java/aqa_hw_10/step_definitions/HomePageStepDefinition.java (hunk 4 above)
Comment (TanyaB777) [reply src/test/java/aqa_hw_10/step_definitions/HomePageStepDefinition.java]: fixed

File: pom.xml (hunk 12 above)
Comment (TanyaB777) [reply pom.xml]: fixed

File: src/main/java/aqa_hw_10/CartPage.java (hunk 14 above)
Comment (TanyaB777) [reply src/main/java/aqa_hw_10/CartPage.java]: fixed

**Pull request merged**